### PR TITLE
OAuth2 Refresh Token Support

### LINKED ISSUE #1498: OAuth2 Refresh Token Support?
Hi, I've been looking into the OAuth server support in CAS (http://jasig.github.io/cas/4.1.x/installation/OAuth-OpenId-Authentication.html), and I now have the basic flow working. For our implementation we need to return refresh tokens in addition to the access token. I've been looking through the docs and code, and I don't see how to enable that. 

Is this currently supported? If not, any pointers on how I could extend things to add this support?

### PR DESCRIPTION
**Pull request by leleuj**
Closes #1498

### CODE REVIEW

File: cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/ticket/accesstoken/DefaultAccessTokenFactory.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
     /** Default instance for the ticket id generator. */
     @NotNull
     @Autowired(required = false)
-    @Qualifier("accessTokenIdGenerator")
+    @Qualifier("refreshTokenIdGenerator")
```
Comment (mmoayyed) [inline cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/ticket/accesstoken/DefaultAccessTokenFactory.java]: You're in `DefaultAccessTokenFactory`. Why is a `refreshTokenIdGenerator` referenced here?

**Comment by mmoayyed** [comment]
Looks good. Thanks!

File: cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/ticket/accesstoken/DefaultAccessTokenFactory.java (hunk 1 above)
Comment (leleuj) [inline cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/ticket/accesstoken/DefaultAccessTokenFactory.java]: Good catch. Will fix and do some renamings for consistency as I realized that the `oAuth` prefix is not necessary for access and refresh tokens.

**Comment by leleuj** [comment]
Updated documentation and names.

**Comment by mmoayyed** [comment]
:+1:

**Pull request merged**